src/NBitcoin/ChainedHeader.cs

```diff
@@ -286,19 +286,24 @@ public ChainedHeader FindAncestorOrSelf(ChainedHeader chainedHeader)
         }
 
         /// <summary>
-        /// Finds the ancestor of this entry in the chain that matches the block hash given.
+        /// Finds the ancestor of this entry in the chain that matches the block hash.
+        /// It will not search lower than the optional height parameter.
         /// </summary>
         /// <param name="blockHash">The block hash to search for.</param>
-        /// <returns>The ancestor of this chain that matches the block hash.</returns>
-        public ChainedHeader FindAncestorOrSelf(uint256 blockHash)
+        /// <param name="height">Optional height to restrict the search to.</param>
+        /// <returns>The ancestor of this chain that matches the block hash, or null if it was not found.</returns>
+        public ChainedHeader FindAncestorOrSelf(uint256 blockHash, int height = -1)
```

Comment:
So now as we have the new implementation, I raise my question again, would it not be better to have default value zero?